code-challenges.test.js

```diff
@@ -10,50 +10,118 @@
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+describe("caps", () => {
+    it("takes in an array of words and returns an array with all the words capitalized.", () => {
+        const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+        // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+        const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+        // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+
+      expect(caps(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+      expect(caps(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+  })
+//   1-ReferenceError: caps is not defined
+//   2-pass
+
 
 // b) Create the function that makes the test pass.
+  //create function called "caps"
+  //add parameter "array" to caps
+  //return an array.map higher order function:
+    // in .map function, offer value as parameter. Return value index 0 capitalized with .toUpperCase built in method.
+    // on same return, add .substring built in method of value to "reattach" other characters from index 1 onward in string to the returned element.
+
+const caps = (array) => {
+    return array.map((value) => {
+        return value[0].toUpperCase() + value.substring(1)
+    })
+}
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
+describe("firstVowelIndex", () => {
+    it("takes in a string and logs the index of the first vowel", () => {
+        const vowelTester1 = "learn"
+        // Expected output: 1
+        const vowelTester2 = "academy"
+        // Expected output: 0
+        const vowelTester3 = "challenges"
+        // Expected output: 2
+
+      expect(firstVowelIndex(vowelTester1)).toEqual(1)
+      expect(firstVowelIndex(vowelTester2)).toEqual(0)
+      expect(firstVowelIndex(vowelTester3)).toEqual(2)
+    })
+  })
+
+//    1-ReferenceError: firstVowelIndex is not defined
+//    2-pass
+
 
 // b) Create the function that makes the test pass.
+  // create function named "firstVowelIndex"
+  // add parameter "string" to firstVowelIndex
+  // within function, create for loop to iterate through each character of string. 
+  // create conditional: if for loop encounters a vowel, return i to output the index of first encountered vowel.
+
+const firstVowelIndex = (string) => {
```

Comment:
Great work in this section!  If you want a stretch, consider how you could use higher-order functions to do this problem.